test/src/main/java/org/apache/accumulo/test/ComprehensiveTableOperationsIT.java

```diff
@@ -0,0 +1,957 @@
+        assertEquals(0, ops.listSplits(table).size());
+      } else {
+        ops.addSplits(table, splits);
+        assertTrue(ops.listSplits(table).containsAll(splits));
+      }
+
+      assertEquals(ops.splitRangeByTablets(table, new Range(), 99).size(),
+          ops.listSplits(table).size() + 1);
+    }
+  }
+
+  @Test
+  public void test_locate() throws Exception {
+    // locate for user tables is tested in LocatorIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(LocatorIT.class.getName()));
+    // locate for METADATA and ROOT tables is tested in ManagerAssignmentIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(ManagerAssignmentIT.class.getName()));
+
+    // basic functionality check for locate on FATE and SCAN_REF tables
+    var fateLocations =
+        ops.locate(AccumuloTable.FATE.tableName(), Collections.singletonList(new Range()));
+    var fateGroupByTablet = fateLocations.groupByTablet().keySet();
+    assertFalse(fateGroupByTablet.isEmpty());
+    fateGroupByTablet.forEach(tid -> {
+      var tabletLoc = fateLocations.getTabletLocation(tid);
+      assertNotNull(tabletLoc);
+      assertTrue(tabletLoc.contains(":"));
+    });
+    var scanRefLocations =
+        ops.locate(AccumuloTable.SCAN_REF.tableName(), Collections.singletonList(new Range()));
+    var scanRefGroupByTablet = scanRefLocations.groupByTablet().keySet();
+    assertFalse(scanRefGroupByTablet.isEmpty());
+    scanRefGroupByTablet.forEach(tid -> {
+      var tabletLoc = scanRefLocations.getTabletLocation(tid);
+      assertNotNull(tabletLoc);
+      assertTrue(tabletLoc.contains(":"));
+    });
+  }
+
+  @Test
+  public void test_getMaxRow_deleteRows() throws Exception {
+    // getMaxRow for user tables is tested in FindMaxIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(FindMaxIT.class.getName()));
+    // getMaxRow not tested for system tables. Test basic functionality here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+    createFateTableRow(userTable);
+    createScanRefTableRow();
+    for (var sysTable : AccumuloTable.values()) {
+      var maxRow =
+          ops.getMaxRow(sysTable.tableName(), Authorizations.EMPTY, null, true, null, true);
+      log.info("Max row of {} : {}", sysTable, maxRow);
+      assertNotNull(maxRow);
+    }
+
+    // deleteRows for user tables is tested in DeleteRowsIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(DeleteRowsIT.class.getName()));
+    // deleteRows not tested for system tables. Test basic functionality here
+    for (var sysTable : AccumuloTable.values()) {
+      switch (sysTable) {
+        case METADATA:
+        case ROOT:
+          // should not be able to delete rows of METADATA or ROOT tables
+          assertThrows(AccumuloException.class,
+              () -> ops.deleteRows(sysTable.tableName(), null, null));
+          break;
+        case FATE:
+          break;
```

Comment:
deleteRows() is allowed for FATE table but probably shouldn't be. Can someone confirm that this is an action that should not be allowed? Not testing for now since deleting our FATE operation from the FATE table when it is in progress will cause problems in the MAC and there is no way to put junk data in the FATE table with the Manager running.

Reply:
Seems like it would be ok to add a hard coded check to FateServiceHandler for this like the root and metadata table.

Kinda related, wondering if we have any test that check the default out of the box permissions of the system tables.  Looking into this question I saw that calling this operation requires the write permission on a tablet.  What this operation does can be done much slower w/ a batch writer.

Reply:
Created https://github.com/apache/accumulo/pull/5491

When testing this, noticed you can call `client.tableOperations.deleteRows` with no problems for FATE and SCAN_REF, but trying to use a batch writer 
(`BatchWriter bw = client.createBatchWriter(AccumuloTable.SCAN_REF.tableName())` for example) you cannot write mutations this way:
`org.apache.accumulo.core.client.MutationsRejectedException: constraint violation codes : []  security codes: {accumulo.scanref(ID:+scanref)=[PERMISSION_DENIED]}  # server errors 0 # exceptions 0`

Reply:
deleteRows on system tables is no longer possible. Test now expects deleteRows to throw an exception.